cloudinit/distros/rhel.py

```diff
@@ -228,3 +245,15 @@ def package_command(self, command, args=None, pkgs=None):
     def update_package_sources(self):
         self._runner.run("update-sources", self.package_command,
                          ["makecache"], freq=PER_INSTANCE)
+
+    def is_reboot_required():
+        maj_version = util.lsb_release()['codename'][0]
+        combined_regex = '|'.join(PACKAGES_REQUIRING_REBOOT[maj_version])
+        history_instance = yum.history.YumHistory()
+        last_transaction = history_instance.last()
```

Comment:
This might miss a need to reboot if multiple transactions take place during the init.